File: src/main/scala/com/scz/flink/source/DataStreamTransformationApp.scala
package com.scz.flink.source

import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
import org.apache.flink.api.scala._

/**
  * Created by shen on 2019/12/22.
  */
object DataStreamTransformationApp {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
//    nonParallelSourceFunction(env)
    parallelSourceFunction(env)
    env.execute("source test")
  }

  def nonParallelSourceFunction(env : StreamExecutionEnvironment): Unit = {
    val nonParallelSource : DataStream[Long] = env.addSource(new CustomNonParallelSourceFunction)
    nonParallelSource.print().setParallelism(1)
  }

  def parallelSourceFunction(env : StreamExecutionEnvironment): Unit = {
    val nonParallelSource : DataStream[Long] = env.addSource(new CustomParallelSourceFunction)
    nonParallelSource.print().setParallelism(2)
  }

}

File: src/main/java/com/scz/flk/stream/DataStreamTransformationApp.java
package com.scz.flk.stream;

import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * Created by shen on 2019/12/22.
 */
public class DataStreamTransformationApp {
    public static void main(String[] args) throws Exception {
        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        DataStreamSource<Long> customDS = env.addSource(new CustomRichParallelSourceFunction());
        customDS.print();
        env.execute("source");
    }
}
